__tests__/app.test.js

```diff
@@ -150,29 +151,125 @@ describe("GET /api/articles/:article_id/comments", () => {
       .get("/api/articles/Banana/comments")
       .expect(400)
       .then((res) => {
-        expect(res.error.text).toEqual('bad request')
+        expect(JSON.parse(res.error.text)).toEqual({msg: 'bad request'})
       });
   });
 });
 
-// describe("POST /api/articles/:article_id/comments", () => {
-//   test("201: comments are successfully posted", () => {
-//     const newComment = {
-//       username: "Mithril_dagger",
-//       body: "All is good here!"
-//     };
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: comments are successfully posted", () => {
+    const newComment = {
+      username: "lurker",
+      body: "All is good here!"
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment) 
+      .expect(201) 
+      .then(({body}) => {
+        
+        expect(Object.keys(body).length).toEqual(6)
+        expect(body.body).toEqual("All is good here!")
+        expect(body.author).toEqual("lurker")
+        expect(body.article_id).toEqual(1)
+        expect(typeof body.article_id && typeof body.votes && typeof body.comment_id).toEqual("number")
+        expect(Object.keys(body)).toEqual(
```

Comment:
You have already asserted that these keys exist by checking their values. What does checking the Object.keys achieve here?